src/mbgl/gl/attributes.hpp

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <mbgl/gl/bytes.hpp>
+
+namespace mbgl {
+namespace gl {
+
+template <typename T> struct AttributeType;
+
+template <> struct AttributeType<int8_t>   : std::integral_constant<GLenum, GL_BYTE> {};
+template <> struct AttributeType<uint8_t>  : std::integral_constant<GLenum, GL_UNSIGNED_BYTE> {};
+template <> struct AttributeType<int16_t>  : std::integral_constant<GLenum, GL_SHORT> {};
+template <> struct AttributeType<uint16_t> : std::integral_constant<GLenum, GL_UNSIGNED_SHORT> {};
+template <> struct AttributeType<int32_t>  : std::integral_constant<GLenum, GL_INT> {};
+template <> struct AttributeType<uint32_t> : std::integral_constant<GLenum, GL_UNSIGNED_INT> {};
+template <> struct AttributeType<float>    : std::integral_constant<GLenum, GL_FLOAT> {};
+template <> struct AttributeType<double>   : std::integral_constant<GLenum, GL_DOUBLE> {};
```

Comment:
At some point, we should hide the `GL_*`, and `gl*` symbols so users of this library don't need to include `gl.h`.